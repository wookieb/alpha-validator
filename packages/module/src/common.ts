import * as is from "predicates";

export function assertValidSchemaNames(schemaNames: string[]) {
    is.assert(is.notEmptyArray, 'You need to provide at least one schema name')(schemaNames);
    is.assert(is.arrayOf(is.notBlank), 'Every schema name must be a non-blank string')(schemaNames);
}