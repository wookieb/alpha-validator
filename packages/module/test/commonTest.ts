import {assertValidSchemaNames} from "@src/common";

describe('common', () => {
    describe('assertValidSchemaNames', () => {
        describe('failure', () => {
            describe('requires schema name', () => {
                it('to be a non-blank string', () => {
                    expect(() => {
                        assertValidSchemaNames(['   '])
                    })
                        .toThrowErrorMatchingSnapshot();
                });

                it('to be a non-empty string', () => {
                    expect(() => {
                        assertValidSchemaNames([''])
                    })
                        .toThrowErrorMatchingSnapshot();
                });

                it('to be a string', () => {
                    expect(() => {
                        assertValidSchemaNames([true] as any)
                    })
                        .toThrowErrorMatchingSnapshot();
                });
            });
        });

        it('success', () => {
            expect(() => {
                assertValidSchemaNames(['schema', 'valid1'])
            })
                .not
                .toThrow();
        });
    });
});