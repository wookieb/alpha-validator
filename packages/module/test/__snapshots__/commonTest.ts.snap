// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common assertValidSchemaNames failure requires schema name to be a non-blank string 1`] = `"Every schema name must be a non-blank string"`;

exports[`common assertValidSchemaNames failure requires schema name to be a non-empty string 1`] = `"Every schema name must be a non-blank string"`;

exports[`common assertValidSchemaNames failure requires schema name to be a string 1`] = `"Every schema name must be a non-blank string"`;
